opengrok-indexer/src/main/java/org/opengrok/indexer/history/RepositoryInfo.java

```diff
@@ -89,6 +90,30 @@ public RepositoryInfo(RepositoryInfo orig) {
         this.currentVersion = orig.currentVersion;
     }
 
+    /**
+     * @return Data Transfer Object for RepositoryInfo.
+     * It will have the same members as in RepositoryInfo except datePatterns.
+     */
+    public Object getRepositoryInfoData() {
```

Comment:
I think it's perhaps unusual for classes to create their own DTOs. For OpenGrok maybe it should be `org.opengrok.indexer.web` that should own the DTO logic, external to e.g. `org.opengrok.indexer.history` or `org.opengrok.indexer.configuration`?

Reply:
Both approaches have some sense I think. Having the DTO definition logic near web app is possible a bit more neat.